tests/sample_size/test_metrics.py

```diff
@@ -76,6 +77,74 @@ def test_boolean_metric_get_probability_too_small(self):
             "Error: Please provide a float between 0 and 1 for probability.",
         )
 
+    @patch("scipy.stats.uniform.rvs")
+    @patch("scipy.stats.nct.rvs")
+    def test_boolean_metric_generate_p_value_two_variants_true_null(self, mock_nct, mock_uniform):
+        true_alt = False
+        size = 200
+        variants = 2
+        pval = 1
+        mock_uniform.return_value = pval
+
+        boolean = BooleanMetric(self.DEFAULT_PROBABILITY, self.DEFAULT_MDE)
+        p = boolean.generate_p_value(true_alt, size, variants)
+        mock_uniform.assert_called_once_with(0, 1, size=variants - 1)
+        mock_nct.assert_not_called()
+        self.assertEqual(p, pval)
+
+    @patch("sample_size.metrics.BooleanMetric.variance")
+    @patch("scipy.stats.uniform.rvs")
+    @patch("scipy.stats.nct.rvs")
+    def test_boolean_metric_generate_p_value_two_variants_true_alt(self, mock_nct, mock_uniform, mock_variance):
+        true_alt = True
+        size = 200
+        variants = 2
+        z = 0
+        pval = 1
+        nc = np.sqrt(size / 2) * self.DEFAULT_MDE / self.DEFAULT_MOCK_VARIANCE
+        mock_variance.__get__ = MagicMock(return_value=self.DEFAULT_MOCK_VARIANCE)
```

Comment:
You can just set this up with

```suggestion
        boolean = BooleanMetric(self.DEFAULT_PROBABILITY, self.DEFAULT_MDE)
        boolean.varaicne = self.DEFAULT_MOCK_VARIANCE
```

Reply:
I got `AttributeError: can't set attribute` when doing this